src/AbstractResourceListener.php

```diff
@@ -180,7 +181,7 @@ public function dispatch(ResourceEvent $event)
                 $id = $event->getParam('id', null);
                 return $this->fetch($id);
             case 'fetchAll':
-                $queryParams = $event->getQueryParams() ?: [];
+                $queryParams = $event->getQueryParams() ?: new Parameters();
```

Comment:
This can be reverted, if the type is expanded below

Reply:
done